lib/config/options/index.ts

```diff
@@ -2348,6 +2348,36 @@ const options: RenovateOptions[] = [
     cli: false,
     env: false,
   },
+  {
+    name: 'certificateAuthority',
+    description: 'The overriding trusted CA certificate',
+    type: 'string',
+    stage: 'repository',
+    parent: 'hostRules',
+    default: null,
+    cli: false,
+    env: false,
+  },
+  {
+    name: 'privateKey',
+    description: 'The private key in PEM format',
+    type: 'string',
+    stage: 'repository',
+    parent: 'hostRules',
+    default: null,
+    cli: false,
+    env: false,
+  },
+  {
+    name: 'certificate',
```

Comment:
`certificate` is a bit generic. It may cause problems later if we need to support another style of certificate. How about calling it `certificateChainInPEMFormat` or something like that? That prevents naming problems later, and the config option explains the wanted input.

Reply:
another style did you mean another format of the certificate?

Reply:
I meant a different format of certificate. 🙂

Reply:
right. but I wonder how likely supporting other format of certificate is gonna happen. 

Given both downstream options [in got](https://github.com/sindresorhus/got/blob/main/documentation/5-https.md#certificate) and [in the underlying node tls.SecureContext](https://nodejs.org/api/tls.html#tlscreatesecurecontextoptions). they are all specified as PEM format , this mean the support to any other certificate format will eventually needed to be converted back to PEM format, this will complicate the currently simple implementation of `applyHostRules` function, which is not doing much for the moment.

Reply:
I think the prefixing these with `mTLS` might be test. e.g. `mTLSCertificateAuthority`, `mTLSPrivateKey`, `mTLSCertificate`. Using `mtls` in lower case might be easier to read but less technically correct?

Reply:
my purpose is `mTLS` this time but the options are more around advanced `https` config as how `got` [options](https://github.com/sindresorhus/got/blob/main/documentation/5-https.md) arranged. probably prefix them with `https` makes more sense than `mTLS`? also if someone needed to feed other https options in the future `https` will also fit.

Reply:
Great point, thanks. `https` as the prefix makes more sense